syscall/syscall_name.c

```diff
@@ -0,0 +1,61 @@
+/****************************************************************************
+ * syscall/syscall_name.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#ifndef CONFIG_LIB_SYSCALL
+#define CONFIG_LIB_SYSCALL
+#endif
+#include <syscall.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/* System call name tables.  This table is indexed by the system call numbers
+ * defined above.  Given the system call number, this table provides the
+ * name of the corresponding system function.
+ */
+
+const char *g_syscallname[SYS_nsyscalls + 1] =
```

Comment:
Who will use g_syscallname?
Should we change the name to g_funcnames and put the declaration into nuttx/include/sys/syscall.h like other variables?

Reply:
It is intended to be used by the application code which gets the data from /dev/note to convert syscall number into its name.
It's ok to change the variable name and put the declaration into syscall.h. But unlike others, it should not be surrounded by `#ifdef __KERNEL__`

Reply:
Yes, just like g_funcnparms without __KERNEL__.